src/gcp/cloudfunctions.js

```diff
@@ -65,13 +65,21 @@ function _createFunction(options) {
   const location = "projects/" + options.projectId + "/locations/" + options.region;
   const func = location + "/functions/" + options.functionName;
   const endpoint = "/" + API_VERSION + "/" + location + "/functions";
+
   const data = {
     sourceUploadUrl: options.sourceUploadUrl,
     name: func,
     entryPoint: options.entryPoint,
     labels: options.labels,
     runtime: options.runtime,
   };
+
+  if (options.vpcConnector) {
+    data.vpcConnector = options.vpcConnector;
+  }
```

Comment:
```suggestion
  if (options.vpcConnector) {
    data.vpcConnector = options.vpcConnector;
    // use implied project/location if only given connector id
    if (!data.vpcConnector.includes("/")) {
      data.vpcConnector = `${location}/connectors/${data.vpcConnector}`;
    }
  }
```

One last update: as part of the internal API review, we want to make the `vpcConnector` something that can be defined in a project-agnostic way, so if it doesn't contain a `/` (i.e. is not of the form `projects/*/locations/*/connectors/*`) we will automatically fill in the project and location based on the current function.

Could you apply the same logic to updates as well?

Reply:
OK. Updated. :+1: